lib/lpc55-rng/src/lib.rs

```diff
@@ -0,0 +1,74 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#![no_std]
+#![no_main]
+
+use core::{cmp, mem};
+use drv_lpc55_syscon_api::{Peripheral, Syscon};
+use drv_rng_api::RngError;
+use lpc55_pac::{pmc, rng, PMC, RNG};
+use rand_core::{impls, Error, RngCore};
+
+/// The Lpc55Rng is a thin wrapper around the LPC55 hardware random number
+/// generator (HRNG).
+pub struct Lpc55Rng {
+    pub pmc: &'static pmc::RegisterBlock,
+    pub rng: &'static rng::RegisterBlock,
+}
+
+impl Lpc55Rng {
+    /// Create a new Lpc55Rng instance after powering on, enabling the clocks
+    /// and reseting the underlying HRNG.
+    pub fn new(syscon: &Syscon) -> Self {
+        let pmc = unsafe { &*PMC::ptr() };
+        pmc.pdruncfg0.modify(|_, w| w.pden_rng().poweredon());
+
+        syscon.enable_clock(Peripheral::Rng);
+        syscon.enter_reset(Peripheral::Rng);
+        syscon.leave_reset(Peripheral::Rng);
+
+        Lpc55Rng {
+            pmc,
+            rng: unsafe { &*RNG::ptr() },
```

Comment:
(same as on line 25)

Reply:
Addressed in [bce3889](https://github.com/oxidecomputer/hubris/pull/1820/commits/bce388971fce1d15496ef0bc7174bc4f4033a5e5) as well.